test/e2e_generic/clusterscope/ingressclasses.go

```diff
@@ -0,0 +1,97 @@
+package clusterscope
+
+import (
+	"context"
+	"time"
+
+	"github.com/loft-sh/vcluster/test/framework"
+	"github.com/onsi/ginkgo"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	IngressClassName           = "vcluster-test-ingress-class"
+	IngressClassControllerName = "vcluster.loft.sh/vcluster-test-ingress-controller"
+)
+
+var _ = ginkgo.Describe("Generic sync cluster scoped resources", func() {
+	f := framework.DefaultFramework
+
+	ginkgo.It("sync cluster scoped resource into vcluster", func() {
+		ctx := context.Background()
+		// create an ingress class in host cluster
+		_, err := f.HostClient.NetworkingV1().IngressClasses().Create(ctx, &networkingv1.IngressClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: IngressClassName,
+			},
+			Spec: networkingv1.IngressClassSpec{
+				Controller: IngressClassControllerName,
+			},
+		}, metav1.CreateOptions{})
+
+		framework.ExpectNoError(err)
+
+		var ingClass *networkingv1.IngressClass
+
+		err = wait.PollImmediate(time.Millisecond*500, framework.PollTimeout, func() (bool, error) {
+			ingClass, err = f.VclusterClient.NetworkingV1().IngressClasses().Get(ctx, IngressClassName, metav1.GetOptions{})
+			if err != nil {
+				if kerrors.IsNotFound(err) {
+					return false, nil
+				}
+
+				return false, err
+			}
+
+			return true, nil
+		})
+
+		// check for ingress class controller name
+		framework.ExpectEqual(ingClass.Spec.Controller, IngressClassControllerName)
+	})
+
+	ginkgo.It("deleting virtual cluster scoped object doesn't delete the physical", func() {
+		ctx := context.Background()
+
+		err := f.VclusterClient.NetworkingV1().IngressClasses().Delete(ctx, IngressClassName, metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
+
+		// should not delete the physical ingress class
+		err = wait.PollImmediate(time.Millisecond*500, framework.PollTimeout, func() (bool, error) {
+			_, err = f.HostClient.NetworkingV1().IngressClasses().Get(ctx, IngressClassName, metav1.GetOptions{})
+			if err != nil {
+				if kerrors.IsNotFound(err) {
+					return false, nil
+				}
```

Comment:
Why is this check required? :thinking:

Reply:
Yea I think it should be the otherway around
```
if kerrors.IsNotFound(err) {
    return false, err   // error for not aligning with assert condition
}

return false, nil   // some other error while fetching, simply retry getting till timeout
```

Reply:
Yes, I agree